src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12354,9 +12390,18 @@ private ParameterSyntax ParseLambdaParameter()
             }
 
             SyntaxToken paramName = this.ParseIdentifierToken();
-            var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
-            exclamation = exclamation is null ? null : CheckFeatureAvailability(exclamation, MessageID.IDS_ParameterNullChecking);
-            var parameter = _syntaxFactory.Parameter(default(SyntaxList<AttributeListSyntax>), modifiers.ToList(), paramType, paramName, exclamation, null);
+            SyntaxToken exclamationExclamation = null;
+            if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+            {
+                exclamationExclamation = MergeTokens(this.EatToken(SyntaxKind.ExclamationToken), this.EatToken(SyntaxKind.ExclamationToken), SyntaxKind.ExclamationExclamationToken);
+            }
+            if (exclamationExclamation != null)
+            {
+                exclamationExclamation = (exclamationExclamation.Kind != SyntaxKind.ExclamationExclamationToken)
+                    ? ConvertToMissingWithTrailingTrivia(exclamationExclamation, SyntaxKind.ExclamationExclamationToken)
+                    : CheckFeatureAvailability(exclamationExclamation, MessageID.IDS_ParameterNullChecking);
```

Comment:
Consider extracting this to a helper method, and using here and above.

Reply:
Hmm, since it's a single line, I don't think we have to.